### PR TITLE
Testclick

### PR DESCRIPTION
**Pull request by devguydavid**
Example of how to fire "real" clicks.

Also, note that we might want to standardize on using aria-label as opposed to the data-testid approach because it will encourage using aria-label which is good for accessibility.

### CODE REVIEW

File: src/navigation/SearchBar.test.js [hunk 1]
```diff
@@ -111,3 +114,19 @@ describe("listener callback tests", () => {
...
+    function handleClick() {
+        clicked = true;
+    }
+    act(() => {
+        render(<SearchBar onSearch={() => handleClick()} />, root);
+    });
+
+    // Click the main menu button
+    //fireEvent.click(screen.getByRole("button", { label: /main-menu-button/i }));
+    fireEvent.click(screen.getByLabelText("search-button"));
```
Comment (devguydavid) [inline src/navigation/SearchBar.test.js:129]: See? Clicking! :)

File: src/navigation/DropMenu.test.js [hunk 2]
```diff
@@ -108,3 +109,26 @@ describe("listener callback tests", () => {
...
+                ariaLabel="test-dropmenu"
+                onClick={() => handleClick()}
+            />,
+            root
+        );
+    });
+
+    // Click the main menu button
+    //fireEvent.click(screen.getByRole("button", { label: /main-menu-button/i }));
+    fireEvent.click(screen.getByLabelText("test-dropmenu"));
```
Comment (devguydavid) [inline src/navigation/DropMenu.test.js:131]: Clicky click!

**Review by skylon07 — DISMISSED** [review]
Awesome! I remembered seeing you made these changes, and actually forgot to do research on the aria-* attributes. So thanks for reminding me ;)

Anyway, I do like the idea of being able to use screen.getBySomething() instead of my grabSomething() functions. I feel like it is more robust, and doesn't rely on the internal structure of the component like my grab()s do. My only concern with the changes is that you're using fireEvent(). I remember reading in the React docs that they "prefer firing real DOM events", although I'm not sure they explained why. It probably is for niche cases which I'm sure only big companies like Facebook would run into consistently enough...
But anyway, now I'm thinking it might be worth it to try and see ourselves, so let's try using fireEvent() and just making real DOM tests for components that don't work well for it, if there even are any. Bottom line is a test is a test, right?

Back to the aria-label thing really quick, is it normal to have an arial-label prop in a component? My thinking was all DropMenus would render the same thing: a menu (that drops). What are the benefits for having a prop dedicated for it? Is it useful for it to be a dynamic label? (I probably just don't understand what aria-label actually represents/does with screen readers.)

Thanks for taking time to write those tests! Since there's nothing incompatible/breaking, and my questions were just design-philosophy based, I'll approve the request and let you merge it from there. You rock!

**Comment by skylon07** [comment]
Because there are conflicts with main, and because I'll be sweeping all files performing the same kinds of changes, I won't merge this into main anymore. However, I've noted the changes you've made, so I'll be able to copy that reasoning into all the files. Thanks for showing me how to do this!

**Pull request closed without merging**